File: src/api/exhibition.ts
import request from '@/utils/request'

export interface Exhibition {
    id: number
    name: string
    shortDesc?: string
    description?: string
    startDate?: string
    endDate?: string
    status?: number
    price?: number
    coverImage?: string
    tags?: string
    // 兼容前端可能的字段名
    date?: string 
}

export const exhibitionApi = {
    // 获取当前主推展览
    getCurrent() {
        return request.get<any, Exhibition>('/exhibition/current')
    },

    // 获取展览列表
    getList(status?: 'ongoing' | 'upcoming') {
        return request.get<any, Exhibition[]>('/exhibition/list', {
            params: { status }
        })
    },

    // 获取详情
    getDetail(id: number) {
        return request.get<any, Exhibition>(`/exhibition/${id}`)
    }
}

File: shared-backend/src/main/java/com/buyticket/entity/Exhibition.java
package com.buyticket.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("exhibition")
public class Exhibition {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    private String name;

    private String shortDesc;

    private String description;

    private LocalDate startDate;

    private LocalDate endDate;

    /**
     * 状态 (0:待开始, 1:进行中, 2:已结束)
     */
    private Integer status;

    private BigDecimal price;

    /**
     * 每小时门票数量
     */
    private Integer ticketsPerHour;

    /**
     * 9-12点时段门票数量
     */
    private Integer morningTickets;

    /**
     * 14-17点时段门票数量
     */
    private Integer afternoonTickets;

    private String coverImage;

    /**
     * 介绍插图（JSON数组格式存储多张图片）
     */
    private String detailImages;

    /**
     * 标签 (逗号分隔)
     */
    private String tags;

    private LocalDateTime createTime;
}
